src/main/scala/ai/privado/utility/Utilities.scala

```diff
@@ -437,4 +437,105 @@ object Utilities {
     })
     semanticList.toList
   }
+
+    def reportUnresolvedMethods(xtocpg: Try[Cpg], filename: String): Unit = {
+    var total = 0
+    var unresolvedSignatures = 0
+    var unresolvedNamespaces = 0
+    var unresolvedSignaturesList = ListBuffer[String]()
+    var unresolvedNamespacesList = ListBuffer[String]()
+
+    val java_unresolved_signature = "(?i)(.*)(unresolved)(signature)(.*)"
+    val java_unresolved_namespace = "(?i)(.*)(unresolved)(namespace)(.*)"
+    val python_unresolved = "(?i)(.*)(unknownfullname)(.*)"
+
+    var unresolved_sig_pattern = python_unresolved
+    if (filename.equals(Constants.JAVA_STATS)) {
+      unresolved_sig_pattern = java_unresolved_signature
+    }
+
+    xtocpg match {
+      case Success(cpg) => {
+        total = cpg.call.methodFullName.l.length
+        unresolvedSignatures = cpg.call.methodFullName(unresolved_sig_pattern).l.length
+        cpg.call.methodFullName(unresolved_sig_pattern).l.map(us => {
+          unresolvedSignaturesList += us.methodFullName + "\n\t" + "Line Number: " + us.lineNumber.get + "\n\t" + "File: " + getFileNameForNode(us)
+        })
+
+        if (filename.equals(Constants.JAVA_STATS)) {
+          unresolvedNamespaces = cpg.call.methodFullName(java_unresolved_namespace).l.length
+          cpg.call.methodFullName(java_unresolved_namespace).l.map(un => {
+            unresolvedNamespacesList += un.methodFullName + "\n\t" + "Line Number: " + un.lineNumber.get + "\n\t" + "File: " + getFileNameForNode(un)
+          })
+        }
+      }
+    }
+
+    val statfile = File(filename)
+    statfile.write("")
+    val divider = "---------------------------------------------------------------------------------------------------------"
+
+    println()
+    println(divider)
+    var statstr = "Total number of function calls: " + total
+    println(statstr)
+    statfile.appendLine(statstr)
+    println()
+    statfile.appendLine("")
+
+    var percentage: Double = 0.0
+
+    statstr = "Calls with unresolved signatures: " + unresolvedSignatures
+    println(statstr)
+    statfile.appendLine(statstr)
+    if (unresolvedSignatures > 0) {
+      percentage = (unresolvedSignatures.toDouble * 100.0) / total.toDouble
+      statstr = percentage + "% of total calls are unresolved"
+      println(statstr)
```

Comment:
@gaurav-gogia Instead of continously printing and appending Line to a file. You can store the report in a variable and in end print it out to console and store the result in file